src/FunctionInfosManager.cpp

```diff
@@ -0,0 +1,111 @@
+#include "FunctionInfosManager.h";
+#include "Debug.h"
+#include "Constants.h"
+
+void FunctionInfosManager::buildFunctionInfo(const TempFileManager& astInfoFile) {
+    TempFileManager functionInfoTempFile = TempFileManager("functionInfoTempFile");
+
+    std::string readASTCommand =
+        "python3 " + AST_FUNCTION_INFO_PASS_LOCATION + " " +
+        astInfoFile.getFileName() + " " + functionInfoTempFile.getFileName();
+
+    int command_exit_status = system(readASTCommand.c_str());
+
+    logout("ran command " << readASTCommand);
+
+    if (command_exit_status != 0) {
+        errs() << "error occurred while building map; early exit\n";
+        std::exit(1);
+    }
+
+    std::ifstream fiFile = functionInfoTempFile.getFileStream();
+
+    std::string line;
+    if (fiFile.is_open()) {
+        while (std::getline(fiFile, line)) {
+            if (line == "") {
+                continue;
+            }
+
+            // example line
+            // FUNCTION=_IO_vfscanf|RETURNS=int|PARAMS|0|_IO_FILErestrict|1|constcharrestrict|2|struct __va_list_tag|3|intrestrict
+
+            auto fiChunks = rlc_util::splitString(line, '|');
+
+            if (fiChunks.size() < 2) {
+                errs() << "error 0 occurred while reading function info: malformed '" << line << "'. early exit\n";
+                std::exit(1);
+            }
+
+            std::string functionName, returnType;
+
+            // FUNCTION=_IO_vfscanf
+            auto functionChunks = rlc_util::splitString(fiChunks[0], '=');
+            if (functionChunks.size() != 2) {
+                errs() << "error 1 occurred while reading function info: bad function declaration '" << line << "'. early exit\n";
+                std::exit(1);
+            }
+            if (functionChunks[0] != "FUNCTION") {
+                errs() << "error 2 occurred while reading function info: function declaration missing \"FUNCTION\" '" << line << "'. early exit\n";
+                std::exit(1);
```

Comment:
given that the error message includes an error code, why not exit with the corresponding code?

Reply:
The error code is kind of an arbitrary number for now. Eventually, I'll make the erroring more standardized and able to produce an error code for every possible error